src/code/z_skelanime.c

```diff
@@ -0,0 +1,1748 @@
+ */
+void SkelAnime_LodDrawLimb(GlobalContext* globalCtx, s32 limbIndex, Skeleton* skeleton, Vec3s* limbDrawTable,
+                           OverrideLimbDraw overrideLimbDraw, PostLimbDraw postLimbDraw, Actor* actor, s32 dListIndex) {
+    SkelLimbEntry* limbEntry;
+    Gfx* dList;
+    Vec3f pos;
+    Vec3s rot;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    s32 pad;
+
+    SysMatrix_StatePush();
+    limbEntry = Lib_PtrSegToVirt(skeleton->limbs[limbIndex]);
+    limbIndex++;
+    rot = limbDrawTable[limbIndex];
+
+    pos.x = limbEntry->translation.x;
+    pos.y = limbEntry->translation.y;
+    pos.z = limbEntry->translation.z;
+
+    dList = limbEntry->displayLists[dListIndex];
+    if ((overrideLimbDraw == NULL) || (overrideLimbDraw(globalCtx, limbIndex, &dList, &pos, &rot, actor) == 0)) {
+        SysMatrix_RotateAndTranslateState(&pos, &rot);
+        if (dList != NULL) {
+            Gfx* polyTemp = gfxCtx->polyOpa.p;
+
+            gSPMatrix(polyTemp, SysMatrix_AppendStateToPolyOpaDisp(globalCtx->state.gfxCtx),
+                      G_MTX_LOAD);
+
+            gSPDisplayList(polyTemp + 1, dList);
+            gfxCtx->polyOpa.p = polyTemp + 2;
+        }
+    }
+
+
+    if (postLimbDraw != NULL) {
+        postLimbDraw(globalCtx, limbIndex, &dList, &rot, actor);
+    }
+
+    if (limbEntry->firstChildIndex != LIMB_DONE) {
+        SkelAnime_LodDrawLimb(globalCtx, limbEntry->firstChildIndex, skeleton, limbDrawTable, overrideLimbDraw,
+                              postLimbDraw, actor, dListIndex);
+    }
+
+    SysMatrix_StatePop();
+
+    if (limbEntry->nextLimbIndex != LIMB_DONE) {
+        SkelAnime_LodDrawLimb(globalCtx, limbEntry->nextLimbIndex, skeleton, limbDrawTable, overrideLimbDraw,
+                              postLimbDraw, actor, dListIndex);
+    }
+}
+
+/*
+ * Draws the Skeleton described by `skeleton` with a level of detail display list indexed by `dListIndex`
+ */
+void SkelAnime_LodDraw(GlobalContext* globalCtx, Skeleton* skeleton, Vec3s* limbDrawTable,
+                       OverrideLimbDraw overrideLimbDraw, PostLimbDraw postLimbDraw, Actor* actor, s32 dListIndex) {
+    SkelLimbEntry* limbEntry;
+    s32 pad;
+    Gfx* dList;
+    Vec3f pos;
+    Vec3s rot;
+    GraphicsContext* gfxCtx;
+    s32 pad2;
+
+    if (skeleton == NULL) {
+        return;
+    }
+
+    gfxCtx = globalCtx->state.gfxCtx;
+
+    SysMatrix_StatePush();
+
+    limbEntry = Lib_PtrSegToVirt(skeleton->limbs[0]);
+    pos.x = limbDrawTable[0].x;
+    pos.y = limbDrawTable[0].y;
+    pos.z = limbDrawTable[0].z;
+
+    rot = limbDrawTable[1];
+    dList = limbEntry->displayLists[dListIndex];
+
+    if ((overrideLimbDraw == NULL) || (overrideLimbDraw(globalCtx, 1, &dList, &pos, &rot, actor) == 0)) {
+        SysMatrix_RotateAndTranslateState(&pos, &rot);
+        if (dList != NULL) {
+            Gfx* polyTemp = gfxCtx->polyOpa.p;
+
+            gSPMatrix(polyTemp, SysMatrix_AppendStateToPolyOpaDisp(globalCtx->state.gfxCtx),
```

Comment:
is gfxCtx->polyOpa.p++ OK here? If not, why not use polyTemp++?

Reply:
polyOpa.p is not saved between instructions, and the instructions are offset by 0, 4, 8 and C, rather than 0/4, 0/4. ptr and ptr + 1 is required for the macros,a and the pointer gets a + 2 after them both. 

Not using a temp and using polyOpa.p and polyOpa.p + 1, polyOpa.p += 2 does not match, I could only match it with a temp variable.

Reply:
Understood. I just wanted to ensure you tried it.